pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -897,9 +897,10 @@ func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client,
 
 	wl := &kueue.Workload{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:  p.pod.GetNamespace(),
-			Labels:     map[string]string{},
-			Finalizers: []string{kueue.ResourceInUseFinalizerName},
+			Namespace:   p.pod.GetNamespace(),
+			Labels:      map[string]string{},
+			Finalizers:  []string{kueue.ResourceInUseFinalizerName},
+			Annotations: p.pod.Annotations,
```

Comment:
Do we want to filter this to just pass the ones with the provisioning request prefix for now? then this logic may be updated when someone wants to use annotations for something else in the future

Reply:
Sure, fixed